scripts/provision_credentials.py

```diff
@@ -123,26 +115,33 @@ def _destroy_aws_secrets_manager_secret(self, secret_name):
                 SecretId=secret_name,
                 ForceDeleteWithoutRecovery=True
             )
-        except ClientError as e:
-            if e.response['Error']['Code'] == 'ResourceNotFoundException':
-                logger.info('AWS secret %s does not exist. No changes will be made.', secret_name)
-            else:
-                raise
+        except self.secrets_manager.exceptions.ResourceNotFoundException:
+            logger.info('AWS secret %s does not exist. No changes will be made.', secret_name)
         else:
             assert response['Name'] == secret_name
-            logger.info('Successfully deleted AWS secret %s.', secret_name)
+            # AWS docs recommend waiting for a "ResourceNotFoundException" because
+            # "The deletion is an asynchronous process. There might be a short
+            # delay". https://aws.amazon.com/premiumsupport/knowledge-center/delete-secrets-manager-secret/
+            start = time.time()
```

Comment:
```suggestion
            deadline = time.time() + 60
```

and ensure that the loop is executed at least once. Don't take time.time() more than once in loop body.

Reply:
per iteration that is